presto-main/src/main/java/com/facebook/presto/operator/aggregation/ApproximateLongArrayPercentileAggregations.java

```diff
@@ -0,0 +1,153 @@
+        checkWeight(weight);
+        addInput(state, valuesArrayBlock, weight, percentile, accuracy);
+    }
+
+    private static void addInput(
+            @AggregationState ApproxPercentileState state,
+            @SqlType("array(bigint)") Block valuesArrayBlock,
+            @SqlType(StandardTypes.BIGINT) long weight,
+            @SqlType(StandardTypes.DOUBLE) double percentile,
+            @SqlType(StandardTypes.DOUBLE) double accuracy)
+    {
+        initializeQuantileDigestArray(state, valuesArrayBlock, accuracy);
+        state.addDigests(valuesArrayBlock, weight);
+        state.setPercentile(percentile);
+    }
+
+    @CombineFunction
+    public static void combine(@AggregationState ApproxPercentileState state, ApproxPercentileState otherState)
+    {
+        QuantileDigest[] otherDigests = otherState.getDigests();
+        QuantileDigest[] digests = state.getDigests();
+
+        if (digests == null) {
+            state.setDigests(otherDigests);
+            state.addMemoryUsage(state.estimatedInMemorySizeInBytes());
+        }
+        else {
+            checkState(otherDigests != null, "The digests to be combined cannot be null");
+            checkState(otherDigests.length == digests.length, "The sizes of inputs cannot be different");
+
+            for (int i = 0; i < digests.length; i++) {
+                state.addMemoryUsage(-digests[i].estimatedInMemorySizeInBytes());
+                digests[i].merge(otherDigests[i]);
+                state.addMemoryUsage(digests[i].estimatedInMemorySizeInBytes());
+            }
+        }
+
+        state.setPercentile(otherState.getPercentile());
+    }
+
+    @OutputFunction("array(bigint)")
+    public static void output(@AggregationState ApproxPercentileState state, BlockBuilder out)
+    {
+        QuantileDigest[] digests = state.getDigests();
+        double percentile = state.getPercentile();
+
+        if (digests == null) {
+            out.appendNull();
+        }
+        else {
+            checkState(percentile != -1.0, "Percentile is missing");
+            checkCondition(0 <= percentile && percentile <= 1, INVALID_FUNCTION_ARGUMENT, "Percentile must be between 0 and 1");
+
+            BlockBuilder blockBuilder = out.beginBlockEntry();
+            for (QuantileDigest digest : digests) {
+                BIGINT.writeLong(blockBuilder, digest.getQuantile(percentile));
+            }
+            out.closeEntry();
+        }
+    }
+
+    private static void initializeQuantileDigestArray(@AggregationState ApproxPercentileState state, Block valuesArrayBlock, double accuracy)
+    {
+        if (state.getDigests() == null) {
+            checkAccuracy(accuracy);
+            int length = valuesArrayBlock.getPositionCount();
+            QuantileDigest[] digests = new QuantileDigest[length];
+            for (int i = 0; i < length; i++) {
+                digests[i] = new QuantileDigest(accuracy);
+            }
+            state.setDigests(digests);
+        }
+    }
```

Comment:
This should be in the state constructor. Look at some of the existing aggs for the style we use.